File: backend/app/schemas/budget.py
from pydantic import BaseModel, Field


class BudgetCreate(BaseModel):
    monthly_limit: float = Field(gt=0)
    month: str
    year: int


class BudgetResponse(BaseModel):
    id: int
    monthly_limit: float
    month: str
    year: int

    class Config:
        from_attributes = True

File: frontend/src/types/budget.ts
export interface Budget {
    id: number;
    monthly_limit: number;
    month: string;
    year: number;
}

export interface CreateBudgetRequest {
    monthly_limit: number;
    month: string;
    year: number;
}

export interface UpdateBudgetRequest {
    monthly_limit: number;
    month: string;
    year: number;
}